ui/static/js/script.js

```diff
@@ -51,13 +41,47 @@ document.getElementById("addressForm").addEventListener("submit", function (even
     fetch("/api/analyze/", {
         method: "POST",
         headers: { "Content-Type": "application/json" },
-        body: JSON.stringify({ "street": street, "houseNumber": houseNumber, "radius": radius }),
+        body: JSON.stringify({ street: street, houseNumber: houseNumber, radius: radius }),
     })
         .then(response => response.json())
         .then(data => {
-            const responseBox = document.getElementById("response");
-            responseBox.classList.add("visible");
-            responseBox.innerText = JSON.stringify(data, null, 2);
+            const devResponseBox = document.getElementById("dev-response");
+            devResponseBox.classList.add("visible");
+            devResponseBox.innerText = JSON.stringify(data, null, 2);
+
+            let addressListHTML = "<div><p>The following addresses are problematic:</p><ul>";
+            data.forEach(entry => {
+                let listItem = "<li>";
+
+                // Loop through each key in the entry object
+                for (const [key, value] of Object.entries(entry)) {
+                    // Add the key-value pair to the list item
+                    const fieldName = key.replace(/_/g, ' ').toUpperCase();  // Optional: clean up field names
+                    const fieldValue = value || `No ${fieldName} provided`;
+
+                    listItem += `${fieldName}: ${fieldValue}, `;
+                }
+
+                // Remove the last comma and space
+                listItem = listItem.slice(0, -2);
```

Comment:
In python you can use `str.join` to avoid these kind of shenanigans :) JS seems to have `Array.prototype.join()` that might work for you.